tests/e2e/tests/simple/simple1_test.go

```diff
@@ -72,50 +66,36 @@ func TestMain(m *testing.M) {
 // TODO: need an "is cluster ready" before running any tests
 
 func TestSimpleIngress(t *testing.T) {
-	// Tests the rewrite/dropping of the /fortio/ prefix as fortio only replies
-	// with "echo debug server ..." on the /debug uri.
+	// Tests that a simple ingress with rewrite/dropping of the /fortio/ prefix
+	// works, as fortio only replies with "echo debug server ..." on the /debug uri.
 	url := tc.Kube.IngressOrFail(t) + "/fortio/debug"
-	log.Infof("Fetching '%s'", url)
-
-	retry := util.Retrier{
-		BaseDelay:   time.Second,
-		MaxDelay:    time.Second,
-		Retries:     1000,
-		MaxDuration: 105 * time.Second,
-	}
-
-	client := &http.Client{Timeout: 5 * time.Second}
-	_, err := retry.Retry(context.Background(), func(ctx context.Context, i int) error {
-		resp, err := ctxhttp.Get(ctx, client, url)
-		if err != nil {
-			log.Warnf("Attempt %d : http.Get error %v", i, err)
-			return fmt.Errorf("attempt %d : http.Get error %v", i, err)
-		}
-
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			log.Warnf("Attempt %d : ReadAll error %v", i, err)
-			return fmt.Errorf("attempt %d : ReadAll error %v", i, err)
-		}
 
-		_ = resp.Body.Close()
-		bodyStr := string(body)
-		if len(bodyStr) == 0 {
-			log.Infof("Attempt %d: reply body is empty", i)
-			return fmt.Errorf("attempt %d: reply body is empty", i)
+	log.Infof("Fetching '%s'", url)
+	needle := []byte("echo debug server up")
+
+	o := fhttp.NewHTTPOptions(url)
+	var (
+		code      int
+		data      []byte
+		dataDebug string
+	)
+	for i := 1; i <= 100; i++ { // Wait max ~100s
```

Comment:
Shouldn't that been done in the setup ? When that test case start don t you expect the ingress and service to be up ?

Reply:
I'll clarify the comment: what happens is we have first the ingress startup time but then we also have to wait for the destination service to be up _and_ the rules to propagate